federation/pkg/kubefed/init/init_test.go

```diff
@@ -392,12 +398,20 @@ func TestCertsHTTPS(t *testing.T) {
 	}
 }
 
-func fakeInitHostFactory(federationName, namespaceName, ip, dnsZoneName, image, dnsProvider string) (cmdutil.Factory, error) {
+func fakeInitHostFactory(federationName, namespaceName, ip, dnsZoneName, image, dnsProvider, etcdPVCapacity string) (cmdutil.Factory, error) {
 	svcName := federationName + "-apiserver"
 	svcUrlPrefix := "/api/v1/namespaces/federation-system/services"
 	credSecretName := svcName + "-credentials"
 	cmKubeconfigSecretName := federationName + "-controller-manager-kubeconfig"
-	capacity, err := resource.ParseQuantity("10Gi")
+	pvCap := ""
+
+	if "" != etcdPVCapacity {
+		pvCap = etcdPVCapacity
+	} else {
```

Comment:
Drop the else and just hoist this outside the condition. This entire thing then looks like:

```go
pvCap := "10Gi"
if etcdPVCapacity != "" {
    pvCap = etcdPVCapacity
}
```